firewalld should fail nicely when service is stopped

willthames opened the pull request:
- Feature Pull Request

##### COMPONENT NAME
firewalld

##### ANSIBLE VERSION
```
ansible 2.2.0 (devel c752b25ced) last updated 2016/09/07 00:54:00 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 5a5c1491ae) last updated 2016/09/07 01:08:07 (GMT -400)
  lib/ansible/modules/extras: (devel 8bfdcfcab2) last updated 2016/09/07 01:09:04 (GMT -400)
  config file = 
  configured module search path = Default w/o overrides
```

##### SUMMARY

If firewalld is not running, it should print out a nice error message suggesting that it should be running, and not complain about potentially missing modules.

Before:

```
$ ansible -m firewalld -a "state=enabled permanent=true" localhost
 [WARNING]: Host file not found: /etc/ansible/hosts

 [WARNING]: provided hosts list is empty, only localhost is available

localhost | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "firewalld and its python 2 module are required for this module"
}
```

After:

```
$ ansible -m firewalld -a "state=enabled permanent=true" localhost
 [WARNING]: Host file not found: /etc/ansible/hosts

 [WARNING]: provided hosts list is empty, only localhost is available

localhost | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "firewalld must be running to use this module"
}
```

vmindru commented on system/firewalld.py:
```diff
@@ -360,6 +357,9 @@ def main():
         module.fail_json(msg='unsupported version of firewalld, requires >= 2.0.11')
 
+    if not fw.connected:
+        module.fail_json(msg="firewalld must be running to use this module")
```
aren't we already doing something similar on 389 ?

vmindru commented:
wow Will :) we just talked about this and you already fired-up a PR :) nice

willthames commented:
@vmindru thanks for the feedback - you are right, the checks were already there. 

The previous problem was that the HAVE_FIREWALLD failures prevented the code ever reaching those checks. However, if you then remove the checks that I added, it fails when it does `fw.getDefaultVersion`, before the check for whether the service is running!

I took the existing checks and moved them to where I added my checks. This should correctly categorise the failures, while not introducing duplicate code.

vmindru commented:
looks good!

gregdek commented:
Thanks @willthames. To the current maintainers, @maxamillion please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

maxamillion commented:
shipit

gregdek commented:
Thanks again to @willthames for this PR, and thanks @maxamillion for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

resmo commented:
This looks more like an "improvement" than a feature to me. Merging.
